src/main/scripts/jbang

```diff
@@ -1,35 +1,91 @@
 #!/usr/bin/env bash
 
+# The Java version to install when it's not installed on the system yet
+javaVersion=11
+
 absolute_path() {
-    # if the given path to the jbang launcher is absolute (i.e. it is either starting with / or a
-    # 'letter:/' when using gitbash on windows) it is returned unchanged, otherwise we construct an absolute path
-    [[ $1 = /* ]] || [[ $1 =~ ^[A-z]:/ ]] && echo "$1" || echo "$PWD/${1#./}"
+  # if the given path to the jbang launcher is absolute (i.e. it is either starting with / or a
+  # 'letter:/' when using gitbash on windows) it is returned unchanged, otherwise we construct an absolute path
+  [[ $1 = /* ]] || [[ $1 =~ ^[A-z]:/ ]] && echo "$1" || echo "$PWD/${1#./}"
 }
 
 resolve_symlink() {
-    if [[ $OSTYPE != darwin* ]]; then minusFarg="-f"; fi
-    sym_resolved=$(readlink ${minusFarg} $1)
+  if [[ $OSTYPE != darwin* ]]; then minusFarg="-f"; fi
+  sym_resolved=$(readlink ${minusFarg} $1)
 
-    if [[ -n $sym_resolved ]]; then
-        echo $sym_resolved
-    else
-        echo $1
-    fi
+  if [[ -n $sym_resolved ]]; then
+    echo $sym_resolved
+  else
+    echo $1
+  fi
 }
 
+case "$(uname -s)" in
+  Linux*)   os=linux;;
+  Darwin*)  os=mac;;
+  CYGWIN*)  os=windows;;
+  MINGW*)   os=windows;;
+  *)        echo "Unsupported Operating System"; exit 1;;
+esac
+
+case "$(uname -p)" in
```

Comment:
on os `uname -m` give the arch...

currently this detects 32 bit as it returns i386.

Reply:
Is this something you would be able to fix/implement?

Reply:
I can; but checking on ubuntu `uname -m` works too ....any reason to use uname -p?

Reply:
never mind; adjusted the code to fallthrough various combiations of `uname -` so it will work more places. just pushed the commit - let me know wyt